nixos/modules/hardware/sensor/iio.nix

```diff
@@ -0,0 +1,26 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+{
+  ###### interface
+
+  options = {
+    hardware.sensor.iio = {
+      enable = mkEnableOption "Enable this option to support IIO sensors.";
```

Comment:
Using `hardware.sensor` makes sense to me. I have no better alternative, anyway.